.github/workflows/canary.yml

```diff
@@ -0,0 +1,58 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - canary
+  pull_request:
+    branches:
+      - canary
+
+jobs:
+  # From https://github.com/marketplace/actions/workflow-run-cleanup-action
+  # This action cleans up previously running instances of a workflow on the same branch.
+  # This accomplishes the task of automatically cancelling CI runs on pushes to the same branch,
+  # which is a common feature in most CI systems but currently not possible with GitHub actions.
+  cleanup-runs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rokroskar/workflow-run-cleanup-action@master
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
Shouldn't `NPM_TOKEN` be also defined here to be able to publish to npm?

Reply:
Have you added the `NPM_TOKEN` to the repo?